my-app/src/App.tsx

```diff
@@ -13,61 +16,53 @@ type User = {
   password: string
 }
 
-type Pages = 'Auth' | 'Reg' | 'Posts'
-
 export const App = () => {
   const [users, setUsers] = useState<User[]>([])
-  const [page, setPage] = useState<Pages>('Reg')
+  const [searchResult, setSearchResult] = useState('')
+
+  const navigate = useNavigate()
```

Comment:
try to separate different logic into different components
here you have auth, navigation, and search